### DISCUSSION TITLE
Twin Macro as an ES module import in Framer

### DISCUSSION THREAD

**Discussion by alanontheweb** [ben-rogerson/twin.macro#607] [Q&A]
Hello @ben-rogerson I thought I would reach out to you since you would know best. Framer is soon releasing NPM support visa JSPM with import maps for ES modules. This feature is in beta, but I have gotten styled components working on it. When I reached out to the Framer team about twin-macro support, this is what they said:

![image](https://user-images.githubusercontent.com/302498/148588755-50da77ad-7328-42a4-8246-5d2955e29328.png)

I'm guessing there is no way to get twin-macro to run on the client side. Maybe something like [twind](https://twind.dev/) would do the trick. But I was wondering if there was a way to develop components built with twin-macro in Storybook and export them via rollup (or other tools) as ES modules in an npm package. Then, instead of importing twin-macro, I would import the minified/compiled/packaged version of the components. Is this even possible or does twin-macro still need to be a dependency with that build step involved?

Thank you

**Comment by ben-rogerson** [comment]
Hey Alan

Yeah, your guess is correct - twin won't be able to run client side.
I've made a [component library example](https://github.com/ben-rogerson/twin.examples/tree/master/component-library-emotion)  that precompiles components with Rollup but I haven't tested them in Framer before.
